### PR TITLE
changes_to_read_credentials_from_vault

### PR DESCRIPTION
**Pull request by gau31rav**
**Description of the change**

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/mongodb/templates/arbiter/statefulset.yaml [hunk 1]
```diff
@@ -155,18 +157,24 @@ spec:
...
                   key: mongodb-root-password
+            {{- end }}
+            {{- if .Values.auth.enabled }}
             - name: MONGODB_REPLICA_SET_KEY
               valueFrom:
                 secretKeyRef:
                   name: {{ include "mongodb.secretName" . }}
                   key: mongodb-replica-set-key
             {{- end }}
+            {{- if .Values.auth.credentialFilePath }}
```
Comment (guru1306) [inline bitnami/mongodb/templates/arbiter/statefulset.yaml]: Can we name as vault credentials? Can we explain it in a read me?

File: bitnami/mongodb/templates/standalone/scripts-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    if [[ -n "$MONGODB_CREDENTIAL_FILE" ]] && [[ -f "$MONGODB_CREDENTIAL_FILE" ]]; then
+        while read -r line; do
+        line=$(echo "$line" | sed 's/: /=/g')
+        declare $line; 
+        done < "$MONGODB_CREDENTIAL_FILE"
+        export MONGODB_ROOT_PASSWORD="$MONGODB_ROOT_PASSWORD"
+        export MONGODB_USERNAME="$MONGODB_USERNAME"
+        export MONGODB_PASSWORD="$MONGODB_PASSWORD"
+        export MONGODB_DATABASE="$MONGODB_DATABASE"
+        export ALLOW_EMPTY_PASSWORD="$ALLOW_EMPTY_PASSWORD"
```
Comment (guru1306) [inline bitnami/mongodb/templates/standalone/scripts-configmap.yaml]: What's the  use of ALLOW_EMPTY_PASSWD?

File: bitnami/mongodb/templates/standalone/scripts-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  {{- $releaseNamespace := include "mongodb.namespace" . }} 
+  setup.sh: |-
+    #!/bin/bash
+    if [[ -n "$MONGODB_CREDENTIAL_FILE" ]] && [[ -f "$MONGODB_CREDENTIAL_FILE" ]]; then
+        while read -r line; do
+        line=$(echo "$line" | sed 's/: /=/g')
+        declare $line; 
+        done < "$MONGODB_CREDENTIAL_FILE"
+        export MONGODB_ROOT_PASSWORD="$MONGODB_ROOT_PASSWORD"
+        export MONGODB_USERNAME="$MONGODB_USERNAME"
```
Comment (guru1306) [inline bitnami/mongodb/templates/standalone/scripts-configmap.yaml:20]: What happens if I just set the root username and password and not the custom username/password?

Comment (gau31rav) [reply bitnami/mongodb/templates/standalone/scripts-configmap.yaml:20]: Only root user gets created

File: bitnami/mongodb/templates/arbiter/statefulset.yaml (hunk 1 above)
Comment (gau31rav) [reply bitnami/mongodb/templates/arbiter/statefulset.yaml]: Done

File: bitnami/mongodb/templates/standalone/scripts-configmap.yaml (hunk 2 above)
Comment (gau31rav) [reply bitnami/mongodb/templates/standalone/scripts-configmap.yaml]: This is used when the auth is disabled and Mongo db is supposed to allow access with empty password. Removed from the new code

**Pull request merged**